client/src/actions/eventActions.js

```diff
@@ -6,8 +6,19 @@ import { correctDatesForKeys } from '../utils/dateFormatting';
 import { apiAction } from './modelActions';
 
 export function createEvent(formData) {
+  const postData = correctDatesForKeys(formData, ['date', 'startTime', 'endTime']);
+  postData.date.set({
```

Comment:
I am not sure I follow this correctly. Looks like `date` is being passed into `correctDatesForKeys` and then immediately overridden. Maybe some room for refactoring here